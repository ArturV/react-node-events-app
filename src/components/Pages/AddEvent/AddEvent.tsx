import axios from "axios";
import { useState } from "react";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";

export const AddEvent = () => {
  const [newEvent, setNewEvent] = useState("");

  const resetInput = () => {
    setNewEvent("");
  };

  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
    event.preventDefault();

    try {
      axios
        .post(
          "http://localhost:5000/add-event",
          {
            name: newEvent,
          },
          {
            headers: {
              Authorization: `Bearer ${localStorage.getItem("accessToken")}`,
            },
          }
        )
        .then(() => {
          alert(`Event added`);
          resetInput();
        });
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <>
      <Typography height="40px" variant="h4" sx={{ m: 4 }}>
        Add Event
      </Typography>
      <Container component="main" maxWidth="xs">
        <Box component="form" onSubmit={handleSubmit} sx={{ mt: 1 }}>
          <TextField
            margin="normal"
            required
            fullWidth
            id="event"
            label="Add Event"
            name="event"
            autoFocus
            aria-required="true"
            value={newEvent}
            onChange={(e) => setNewEvent(e.target.value)}
          />

          <Button
            type="submit"
            fullWidth
            variant="contained"
            sx={{ mt: 3, mb: 2 }}
          >
            Add
          </Button>
        </Box>
      </Container>
    </>
  );
};
