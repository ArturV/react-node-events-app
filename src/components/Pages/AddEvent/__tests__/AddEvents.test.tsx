import { render, fireEvent } from "@testing-library/react";
import React from "react";
import axios from "axios";
import { AddEvent } from "../AddEvent";
import renderer from "react-test-renderer";

jest.mock("axios");

describe("AddEvent", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render properly", () => {
    const tree = renderer.create(<AddEvent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
