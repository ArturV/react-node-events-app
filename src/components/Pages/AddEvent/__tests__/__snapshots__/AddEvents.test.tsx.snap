// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEvent should render properly 1`] = `
Array [
  <h4
    className="MuiTypography-root MuiTypography-h4 css-1lq2zml-MuiTypography-root"
  >
    Add Event
  </h4>,
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
  >
    <form
      className="MuiBox-root css-164r41r"
      onSubmit={[Function]}
    >
      <div
        aria-required="true"
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="event"
          id="event-label"
        >
          Add Event
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="event"
            name="event"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Add Event
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Add
      </button>
    </form>
  </main>,
]
`;
